### ISSUE TITLE
Stop pretending input == output formats

### ISSUE THREAD

**Issue by alerque** [kislyuk/yq#58]
The current code makes the assumption that input and output formats are always the same.

I think it would be much more useful if the input and output stages could be separated and options mixed or matched at will. This would turn the project into a general purpose transcoder as well as a query engine for each format. It already does all the transcoding anyway, It seems like an arbitrary restriction to force transcoding back to the same format it came from.

My fix for passthrough JSON input in #57 already introduces one specific scenario of mismatched formats being useful, but a more general purpose solution where input and output formats can be set independently would add a lot more flexibility.

**Comment by kislyuk** [comment]
The code makes no such assumption.

The solution to your original problem is to pass the multiple JSON files as separate arguments: `yq file1.json file2.json` instead of concatenating them all to stdin.

**Comment by alerque** [comment]
I understand there is a workaround for my JSON problem. This issue is not about that at all and I feel like you have acted somewhat dismissively in closing it before understanding the issue.

In trying make a more elegant solution to that issue I reviewed the code, and in doing so I realized there wasn't a general purpose way to map between formats, such as reading YAML, parsing with a jq query, and output XML. Or to query TOML and output YAML.
